distributed/cli/dask_worker.py

```diff
@@ -373,7 +373,10 @@ def run():
 
     try:
         loop.run_sync(run)
-    except (KeyboardInterrupt, TimeoutError):
+    except TimeoutError:
+        # We already log the exception in nanny / worker. Don't do it again.
+        raise TimeoutError("Timed out starting worker.") from None
```

Comment:
Whoa, `raise from`?  New syntax for me.

Reply:
Welcome to the brave new world of Python 3 :)

`raise None` specifically is useful when you think the original traceback isn't helpful, which I think it isn't here. Otherwise, you can `raise from e` if you want to include it.